adsft/extraction.py

```diff
@@ -373,26 +375,35 @@ def __init__(self, dict_item):
 
     def open_xml(self):
         """
-        Opens the XML file and encodes it into UTF-8. Removes some text that
-        has no relevance for XML files, such as HTML tags.
+        Opens the XML file and reads raw string, uses soupparse to encode.
 
+        Removes some text that has no relevance for XML files, such as HTML tags, LaTeX entities
+
+        Note that soupparser.fromstring is called by both open_xml and parse_xml.  
+        open_xml uses soupparser.fromstring because the html and LaTex cleanup needs a string, not a parse tree
+        
         :return: semi-parsed XML content
         """
         raw_xml = None
 
         try:
             logger.debug('Opening the file: {0}'.format(self.file_input))
 
-            import codecs
-
-            with codecs.open(self.file_input, 'r', encoding='utf-8') as f:
-                raw_xml = f.read()
+            with open(self.file_input, 'r') as fp:

```

Comment:
but this will interpret the bytes; are you really really sure that beautiful soup can detect correct encoding from the string you have thus read in? i'd think that the file needs to be read as binary 'b'

Reply:
binary 'b' added